test/templates/template_proxy_request_test.rb

```diff
@@ -0,0 +1,73 @@
+require 'test_helper'
+require 'uri'
+require 'net/http'
+require 'mocha'
+require 'templates/templates_plugin'
+require 'templates/template_proxy_request'
+require 'webmock/test_unit'
+
+
+class TemplateProxyRequestTest < Test::Unit::TestCase
+  def setup
+    @foreman_url = 'https://foreman.example.com'
+    Proxy::SETTINGS.stubs(:foreman_url).returns(@foreman_url)
+    @template_url = 'http://proxy.lan:8443'
+    Proxy::Templates::Plugin.settings.stubs(:template_url).returns(@template_url)
+    @request_env = {
+      'REMOTE_ADDR' => '1.2.3.4'
+    }
+  end
+
+  def text_header_extraction_empty
```

Comment:
`test` is misspelled here and in two other tests below, which means they'll be skipped...